test/e2e/service_externalip_test.go

```diff
@@ -737,3 +800,23 @@ func (data *TestData) waitForServiceConfigured(service *v1.Service, expectedExte
 	}
 	return service, assignedNode, nil
 }
+
+func (data *TestData) waitForServiceLoadBalancerIP(service *v1.Service, expectedLoadBalancerIP string) error {
+	// Do not poll immediate to avoid false negative when the expected IP is empty.
+	return wait.PollUntilContextTimeout(context.Background(), 500*time.Millisecond, 5*time.Second, false, func(ctx context.Context) (done bool, err error) {
+		service, err = data.clientset.CoreV1().Services(service.Namespace).Get(context.TODO(), service.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if expectedLoadBalancerIP == "" {
+			if len(service.Status.LoadBalancer.Ingress) > 0 {
+				return false, err
+			}
+		} else {
+			if len(service.Status.LoadBalancer.Ingress) == 0 || service.Status.LoadBalancer.Ingress[0].IP != expectedLoadBalancerIP {
```

Comment:
should we fail immediately with a hand-crafted error if `expectedLoadBalancerIP != ""` and `service.Status.LoadBalancer.Ingress[0].IP != expectedLoadBalancerIP`?

Reply:
The current code can also be used to validate if changing requested IP works, though we don't have this test case yet.
I agree failing immediately under that condition can help detect some transient issue, but if that could happen, it may also happen that the IP is changed from expected IP to unexpected IP, and returning true immedialy when expected IP is seen is also not enough..